### ISSUE TITLE
[FEA]Groupby DataFrame UDF enhancement: support string and date type, handle null, argument takes variables

### ISSUE THREAD

**Issue by liuyangc1** [rapidsai/cudf#6756]
1. UDF with cuDF doesn't support string or date type currently, it could be helpful this can be added. It says this is being planned, it would be helpful to know approximately when this will be added. 

2. I am having trouble handling null with Groupby Dataframe UDF. I noticed that there is a workaround to handle null with Numba.cuda kernel, but this doesn't seems to work with Groupby Dataframe UDF. 

3. Groupby DataFrame UDF arguments don't take variables. Users have to specify exact column name in the argument and inside the function, take the function from the documentation as an example, the input data has to have a column called 'a' to be able to apply the below function; if we want to use the below function for column 'b', there will be an error. This makes it hard to apply the same function on multiple variables. my current workaround is to rename the column I am working on each time to whatever is used in the UDF, then rename it back. So it could be helpful to allow Groupby DataFrame UDF arguments to take variables.
```
def rolling_avg(e, rolling_avg_e):
    win_size = 3
    for i in range(cuda.threadIdx.x, len(e), cuda.blockDim.x):
        if i < win_size - 1:
            # If there is not enough data to fill the window,
            # take the average to be NaN
            rolling_avg_e[i] = np.nan
        else:
            total = 0
            for j in range(i - win_size + 1, i + 1):
                total += e[j]
            rolling_avg_e[i] = total / win_size
```

**Comment by randerzander** [comment]
Still a valid [FEA]

**Comment by vyasr** [comment]
This issue has a number of different components, some of which have been addressed now while others have not:
- Support for string UDFs is in progress and should be largely available in 22.12/23.02
- Discussion of datetime UDFs is also occurring in #938.
- Null support in UDFs has come a long with. @brandon-b-miller are there still gaps that are relevant to the original request in this issue, or have we addressed them all at this point?
- It sounds like what was originally requested regarding "taking variables" could be addressed using the `*args` of an `apply` function, but I don't know if that will work to dynamically select a column by name.

CC @beckernick in case you have more insight into some of these items. I think this issue should probably be split into a few indicating the remaining feature requests at this point since it has been partially addressed.

**Comment by vyasr** [comment]
I'm going to close this since I think all of the components are now supported. However, since some of the initial requests are somewhat broad, there may be specific subcases that are not yet possible. I'd be happy to discuss those in new issues if they can be more precisely identified.
